plugin/src/main/java/io/jenkins/plugins/analysis/warnings/tasks/TaskScanner.java

```diff
@@ -26,12 +26,37 @@
 
 import io.jenkins.plugins.analysis.core.util.LocalizedSeverity;
 
+
 /**
  * Scans a given input stream for open tasks.
  *
  * @author Ullrich Hafner
  */
 class TaskScanner {
+    class IgnoreSection {
+        private static final String IGNORE_BEGIN = " task-scanner-ignore-begin";
+        private static final String IGNORE_END = " task-scanner-ignore-end";
+
+        private boolean ignore = false;
+
+        public boolean matches(final String line) {
+            if (line.contains(IGNORE_BEGIN)) {
+                ignore = true;
+            }
+            else if (line.contains(IGNORE_END)) {
+                ignore = false;
+            }
+
+            return ignore;
+        }
+
+        public void reset() {
+            ignore = false;
+        }
+    }
+
+    private final IgnoreSection inIgnoreSection = new IgnoreSection();
```

Comment:
Can't this be a local variable in the parse method? Otherwise parallel scans will cause problems.

Reply:
Sure, of course. To be honest, I was not aware that one TaskScanner instance is used for all jobs. Will make it a local variable -and get rid of `reset()` as it is no longer required.